setup.py

```diff
@@ -0,0 +1,54 @@
+# make code as python 3 compatible as possible
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import subprocess
+import setuptools
+import os.path
+import sys
+
+from setuptools.command.develop import develop
+from setuptools.command.install import install
+
+
+# Build anltr files on installation
+#   this is such a mess... it looks like there are
+#   no common steps to develop and install
+
+class AntlrDevelopCommand(develop):
+    def run(self):
+        compile_grammar()
+        develop.run(self)
+
+class AntlrInstallCommand(install):
+    def run(self):
+        compile_grammar()
+        install.run(self)
+
+def compile_grammar():
+    here = os.path.dirname(__file__) or '.'
+    package_dir = os.path.join(here, 'latex2sympy')
+    subprocess.check_output(['antlr4',  'PS.g4', '-o', 'gen'], cwd=package_dir)
```

Comment:
Not sure what a "normal" install of this would be like, but I ended up needing an `antlr.cmd` wrapper, so I had to do some [platform checking](https://github.com/bollwyvl/staged-recipes/blob/latex2sympy/recipes/latex2sympy/setup.py.patch#L8).

Reply:
I was going to suggest adding this in all cases but `antlr.cmd` appears to not be a pip package:

```
> date; pip search antlr.cmd 
Mon  4 Dec 17:10:08 CET 2017
```

Reply:
It's good having a window user to test this, you have my sympathy :). 

On debian antlr looks like this: (https://packages.debian.org/stretch/all/antlr4/filelist).

Reply:
> How hard is it to install antlr on windows? How is antlr installed on your machine?

Well, I built `antlr4` as a conda package on osx/linux/win... and by "built", i mean dumped the `.jar` in place and made a wrapper script, rather than diving into `maven` witchcraft. Usually, this is acceptable for java-based commands... can't speak to anyone using `conda` as a java package manager. I'm not really a windows user (just used a VM), but The Conda Way is to, whenever possible, take care of that community, as people do Get Work Done on windows.

> Conda on linux calls the executable

The linked version on conda-forge (2.7.7) is likely different enough to require a new package anyway. I didn't even test it, since the docs said `antlr4`.

> requiring people to install antlr could prevent adoption.

Anyhoo... in testing, once the the packages are built, they just install, no problem. 

But the bigger story is that `antlr4` isn't even required at runtime of `latex2sympy`, just the runtimes (also built as conda packages). This means it doesn't strictly even _need_ to be in `conda`, and indeed, `include_package_data` works with wheels as well (though I haven't tested that route). Instead of checking the `gen` folder in, just distributing it in your package is just fine, shifting the burden to release time rather than every-user-install time, which is usually a really good trade.